service/kernel/oneapi/select_indexed.h

```diff
@@ -33,28 +32,138 @@ namespace internal
 {
 namespace selection
 {
-class QuickSelectIndexed
+class SelectIndexed
 {
 public:
     struct Result
     {
         UniversalBuffer values;
         UniversalBuffer indices;
 
-        Result(ExecutionContextIface & context, uint32_t K, uint32_t nVectors, TypeId valueType, TypeId indexType, services::Status * status)
-            : values(context.allocate(valueType, nVectors * K, status)), indices(context.allocate(indexType, nVectors * K, status))
+        Result() {}
+        Result(ExecutionContextIface & context, uint32_t K, uint32_t nVectors, TypeId valueType, services::Status * status)
+            : values(context.allocate(valueType, nVectors * K, status)), indices(context.allocate(TypeIds::id<int>(), nVectors * K, status))
         {}
     };
+    struct Params
+    {
+        Params(uint32_t K, TypeId fptype, uint32_t size, daal::algorithms::engines::EnginePtr eng) : nK(K), type(fptype), dataSize(size), engine(eng)
+        {}
+        uint32_t nK;
```

Comment:
Parameters `K` and `nK` are ambiguous here and related mostly to KNN, not to selection params. Can we rename them?